packages/bitcore-client/src/storage.ts

```diff
@@ -73,20 +84,22 @@ export class Storage {
 
   async listWallets() {
     let passThrough = new PassThrough();
-    for (let db of this.db) {
+    const dbs = await this.verifyDbs(this.db);
+    for (let db of dbs) {
       const listWalletStream = await db.listWallets();
       passThrough = listWalletStream.pipe(passThrough, { end: false });
-      listWalletStream.once('end', () => this.db.length-- === 0 && passThrough.end());
+      listWalletStream.once('end', () => dbs.length-- === 0 && passThrough.end());
```

Comment:
this is a typo as `dbs.length === 0` will be evaluated before decrementing. It should be `--dbs.length` as seen below in `listKeys()`